### PR TITLE
add feature for serdes

### PR DESCRIPTION
**Pull request by gwbres**
Hello @lkolbly,
I'm building an application that involves an `ublox` and I need the user to control and parametrize it.
Having the `serdes` capacity would really help. With this commit I introduce a new `use-serde` build feature and some serdes derivation. I am mostly interested in `GpsFix`, `Odo` options and `Ant` options

### CODE REVIEW

File: ublox/Cargo.toml [hunk 1]
```diff
@@ -12,12 +12,14 @@ readme = "../README.md"
 default = ["std"]
 std = []
 alloc = []
+use-serde = ["serde"]
```
Comment (reitermarkus) [inline ublox/Cargo.toml:15]: You can just leave this out, I think there will automatically be a `serde` feature for the optional dependency.

Comment (gwbres) [reply ublox/Cargo.toml]: Hello @reitermarkus,
I confirm, this works

File: ublox/src/lib.rs [hunk 2]
```diff
@@ -69,6 +69,10 @@
 #[cfg(feature = "alloc")]
 extern crate alloc;
 
+#[cfg(feature = "use-serde")]
```
Comment (reitermarkus) [inline ublox/src/lib.rs:72]: The feature should be called `serde` instead of `use-serde`.

Comment (gwbres) [reply ublox/src/lib.rs]: woops, I forgot this, implied by your 1st review,
good catch

**Pull request merged**